modules/users/client/components/utils/networks.js

```diff
@@ -0,0 +1,35 @@
+/**
+  * Determine if given user handle for Warmshowers is an id or username
+  * @link https://github.com/Trustroots/trustroots/issues/308
+  */
+function isWarmshowersId(profile) {
+  let x;
+  return isNaN(profile.extSitesWS) ? !1 : (x = parseFloat(profile.extSitesWS), (0 | x) === x);
```

Comment:
Can you explain what's happening here? I have hard time understanding it.
1. does writing `!1` instead of `false` bring some improvement?
2. `(x = parseFloat(profile.extSitesWS), (0 | x) === x)` what does this code do? I see bitwise operators, assignment of something to x etc. and I don't understand.

Reply:
I copied that function from the previous place without editing it.
As I understand !1 is just for making it shorter (or maybe faster too?).
(0 | x ) removes decimal part so "(0 | x) === x" makes that code return false if what we gave to it(the WS id) was a decimal number.
Although it took me some time to understand it.
Maybe we could change it to something like:

```
function isWarmshowersId(profile) {
  // if it is a number
  if (!isNaN(profile.extSitesWS)){
    const x = parseFloat(profile.extSitesWS);
    // false for decimal number
    return Math.floor(x) === x;
  }
  return false;
}
```

Reply:
Sorry, this is probably my bad and I have no memory of it anymore! 😆 

WS has Drupal user id _and_ usernames — for some folks links to their profile work with username, but not for everyone (they switched it on or off at some point). ID works for everyone.

So I think the goal is to just recognise if user entered numerical ID (that could start with 0 and that cannot be negative or have decimals). Unless I'm forgetting something important. 😅  

Simple regepx to check only for `0-9` would prolly work?

Reply:
@agatatalita Thank you for explaining! I learned something new.